docs/collector.gpu.md

```diff
@@ -0,0 +1,143 @@
+# gpu collector
+
+The gpu collector exposes metrics about GPU usage and memory consumption, both at the adapter (physical GPU) and
+per-process level.
+
+|                     |                                      |
+|---------------------|--------------------------------------|
+| Metric name prefix  | `gpu`                                |
+| Data source         | Perflib                              |
+| Counters            | GPU Engine, GPU Adapter, GPU Process |
+| Enabled by default? | No                                   |
+
+## Flags
+
+None
+
+## Metrics
+
+These metrics are available on supported versions of Windows with compatible GPUs and drivers:
+
+### Adapter-level Metrics
+
+| Name                                         | Description                                              | Type  | Labels |
+|----------------------------------------------|----------------------------------------------------------|-------|--------|
+| `windows_gpu_adapter_memory_committed_bytes` | Total committed GPU memory in bytes per physical GPU     | gauge | `phys` |
+| `windows_gpu_adapter_memory_dedicated_bytes` | Dedicated GPU memory usage in bytes per physical GPU     | gauge | `phys` |
+| `windows_gpu_adapter_memory_shared_bytes`    | Shared GPU memory usage in bytes per physical GPU        | gauge | `phys` |
+| `windows_gpu_local_adapter_memory_bytes`     | Local adapter memory usage in bytes per physical GPU     | gauge | `phys` |
+| `windows_gpu_non_local_adapter_memory_bytes` | Non-local adapter memory usage in bytes per physical GPU | gauge | `phys` |
+
+### Per-process Metrics
+
+| Name                                         | Description                                     | Type  | Labels                          |
+|----------------------------------------------|-------------------------------------------------|-------|---------------------------------|
+| `windows_gpu_engine_time_seconds`            | Total running time of the GPU engine in seconds | gauge | `phys`, `eng`, `engtype`, `pid` |
```

Comment:
```suggestion
| `windows_gpu_engine_time_seconds`            | Total running time of the GPU engine in seconds | gauge | `phys`, `eng`, `engtype`, `process_id` |
```


Using `process_id` instead `pid` is important here. Because would allows us to enrich the query with information from the process collector.

```
windows_gpu_engine_time_seconds * on(process_id) group_left(owner, cmdline) windows_process_info
```

Maybe we can add the query to Useful Queries as well.